arrows/core/example_detector.cxx

```diff
@@ -110,8 +114,12 @@ set_configuration(vital::config_block_sptr config_in)
   d->m_center_y     = config->get_value<double>( "center_y" );
   d->m_height       = config->get_value<double>( "height" );
   d->m_width        = config->get_value<double>( "width" );
-  d->m_dx           = config->get_value<double>( "dx" );
-  d->m_dy           = config->get_value<double>( "dy" );
+  if (config->has_value("dx"))
+    d->m_dx           = config->get_value<double>( "dx" );
+  if (config->has_value("dy"))
+    d->m_dy           = config->get_value<double>( "dy" );
+  if (config->has_value("detections_per_image"))
+    d->m_detections_per_image = config->get_value<int>( "detections_per_image");
```

Comment:
You can avoid using has_value() by supplying a default value. e.g. 
    d->m_width = config->get_value<double>( "width", d->m_width );
I recall seeing a branch of this algorithm with default values in these calls, but it is apparently not in master. Adding the defaults makes for cleaner code.